app/src/main/java/com/p2p/presentation/impostor/create/CreateImpostorFragment.kt

```diff
@@ -0,0 +1,81 @@
+package com.p2p.presentation.impostor.create
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.p2p.R
+import com.p2p.databinding.FragmentCreateImpostorBinding
+import com.p2p.presentation.basegame.BaseGameFragment
+import com.p2p.presentation.impostor.ImpostorViewModel
+import com.p2p.utils.value
+
+class CreateImpostorFragment : BaseGameFragment<
+        FragmentCreateImpostorBinding,
+        ImpostorCreateEvents,
+        CreateImpostorViewModel,
+        ImpostorViewModel>() {
+
+    override val gameViewModel: ImpostorViewModel by activityViewModels()
+    override val viewModel: CreateImpostorViewModel by viewModels()
+
+    override val gameInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentCreateImpostorBinding =
+        FragmentCreateImpostorBinding::inflate
+
+    override fun initUI() {
+        super.initUI()
+        gameViewModel.startConnection()
+        gameBinding.startButton.setOnClickListener {
+            val keyWord = gameBinding.textField.value()
+            viewModel.tryStartGame(keyWord)
+        }
+    }
+
+    override fun setupObservers() {
+        super.setupObservers()
+        with(gameViewModel) {
+            observe(players) {
+                val otherPlayers = otherPlayers()
+                updateConnectedPlayers(otherPlayers)
+                viewModel.updatePlayers(otherPlayers)
+            }
+        }
+    }
+
+    private fun updateConnectedPlayers(players: List<String>?) {
+        val text = if (players?.isNotEmpty() == true)
+            players.joinToString(", ")
+        else resources.getString(R.string.lobby_no_players_yet)
```

Comment:
Por si te interesa, una forma que me parece a mí más linda es:
```suggestion
        val text = players
            ?.takeUnless { it.isEmpty() }
            ?.joinToString()
            ?: resources.getString(R.string.lobby_no_players_yet)
```

Reply:
hermoso